bigquery/main.go

```diff
@@ -283,17 +283,27 @@ func (bq *actualBQ) EnsureDataset(ctx context.Context, datasetName string) error
 func (bq *actualBQ) EnsureTable(ctx context.Context, tableName string) error {
 	// Check for existence of table, create if false
 	bq.table = bq.dataset.Table(tableName)
-	_, err := bq.dataset.Table(tableName).Metadata(ctx)
+	schema, err := bigquery.InferSchema(bqRow{})
 	if err != nil {
-		log.Warningf("Error obtaining table metadata: %v", err)
-		schema, err := bigquery.InferSchema(bqRow{})
-		if err != nil {
-			return fmt.Errorf("Failed to infer schema: %v", err)
-		}
+		return fmt.Errorf("Failed to infer schema: %v", err)
+	}
+	metadata, err := bq.dataset.Table(tableName).Metadata(ctx)
+	if err != nil {
+		log.Warningf("Error obtaining table metadata: %v\nCreating new BigQuery table: %v", err, tableName)
 		// Create table if it does not exist.
 		if err := bq.table.Create(ctx, &bigquery.TableMetadata{Name: tableName, Description: "BigQuery Notifier Build Data Table", Schema: schema}); err != nil {
 			return fmt.Errorf("Failed to initialize table %v: ", err)
 		}
+	} else {
```

Comment:
Do you need this else here? Could the `len` check below be out-dented?

Reply:
outdented the len check to form an else-if, but the else guarantees that the table metadata exists